Scripts/Knight.gd

```diff
@@ -68,14 +68,23 @@ func finish_rolling() -> void:
 
 func finish_attacking() -> void:
 	is_attacking = false
+	can_damage = false
+	attacks_left = 0
```

Comment:
It seems like you don't actually check the value of attacks_left anywhere. You simply set it to 1 at some point and then later set it to 0. Why do you need the variable?